### DISCUSSION TITLE
Periodically can't connect to MQTT broker after upgrade to 12.x

### DISCUSSION THREAD

**Discussion by SteveDinn** [arendst/Tasmota#17292] [Q&A]
### PROBLEM DESCRIPTION
Upgraded most of my devices to v12.2.0, and strangely only in a few cases, devices have become "unavailable" in Home Assistant for a few minutes at a time before coming back. I can tell that these devices are able to reconnect to wi-fi immediately because I can still connect to the web console right after they become "unavailable". The problem is that they can't seem to reconnect to the MQTT broker. Nothing has happened to the MQTT broker because the plethora of other devices (TASMOTA included) in my home network are still using it the whole time.

The one thing I am doing that others may not be is using a DNS name instead of an IP address for my MQTT broker. It is provided by my local DNS server (Pi-Hole), which is up the entire time as well. I've verified that I can lookup the DNS name of the MQTT broker while TASMOTA is having this problem.

Here's what TASMOTA's console messages look like when I connect to the web console (I have redacted personal information):
```lua
15:13:51.914 WIF: Connecting to AP1 WiFi_SSID Channel 4 BSSId [WiFi_BSSID] in mode 11n as MonitorLight...
15:13:51.919 MQT: Attempting connection...
15:13:52.065 MQT: Connect failed to mqtt.example.com:1883, rc -5. Retry in 10 sec
15:14:05.832 MQT: Attempting connection...
15:14:06.046 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 20 sec
15:14:26.833 MQT: Attempting connection...
15:14:26.847 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 30 sec
15:14:57.822 MQT: Attempting connection...
15:14:57.839 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 40 sec
15:15:01.184 RSL: STATE = {"Time":"2022-12-02T15:15:01","Uptime":"31T01:05:55","UptimeSec":2682355,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":44,"POWER":"OFF","Wifi":{"AP":1,"SSId":"WiFi_SSID","BSSId":"[WiFi_BSSID]","Channel":4,"Mode":"11n","RSSI":70,"Signal":-65,"LinkCount":39,"Downtime":"0T00:02:22"}}
15:15:38.839 MQT: Attempting connection...
15:15:38.850 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 50 sec
15:16:30.113 MQT: Attempting connection...
15:16:30.228 MQT: Connected
15:16:30.232 MQT: tele/MonitorLight/LWT = Online (retained)
15:16:30.235 MQT: cmnd/MonitorLight/POWER = 
```
And as shown, it eventually does reconnect, but it takes a few minutes.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff Basic and Sonoff S20
- [x] Tasmota binary firmware version number used: v12.2.0
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: OTA upgrade
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
15:41:49.515 CMD: Backlog Template; Module; GPIO 255
15:41:49.517 SRC: WebConsole from 10.1.1.116
15:41:49.519 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 26, Pld -99, Data 'Template; Module; GPIO 255'
15:41:49.588 SRC: Backlog
15:41:49.591 CMD: Grp 0, Cmd 'TEMPLATE', Idx 1, Len 0, Pld -99, Data ''
15:41:49.597 MQT: stat/MonitorLight/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
15:41:49.801 SRC: Backlog
15:41:49.804 CMD: Grp 0, Cmd 'MODULE', Idx 1, Len 0, Pld -99, Data ''
15:41:49.809 MQT: stat/MonitorLight/RESULT = {"Module":{"1":"Sonoff Basic"}}
15:41:50.052 SRC: Backlog
15:41:50.055 CMD: Grp 0, Cmd 'GPIO', Idx 1, Len 3, Pld 255, Data '255'
15:41:50.067 MQT: stat/MonitorLight/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
15:30:36.503 MQT: stat/MonitorLight/STATUS = {"Status":{"Module":1,"DeviceName":"Monitor Light","FriendlyName":["Monitor Light"],"Topic":"MonitorLight","ButtonTopic":"0","Power":0,"PowerOnState":0,"LedState":0,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":1,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
15:30:36.511 MQT: stat/MonitorLight/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://tasmota.example.com/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"31T01:21:30","StartupUTC":"2022-11-01T13:09:06","Sleep":50,"CfgHolder":4617,"BootCount":2949,"BCResetTime":"2020-04-17T18:36:47","SaveCount":4550,"SaveAddress":"F6000"}}
15:30:36.515 MQT: stat/MonitorLight/STATUS2 = {"StatusFWR":{"Version":"12.2.0(tasmota)","BuildDateTime":"2022-10-17T08:39:23","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"424/699"}}
15:30:36.521 MQT: stat/MonitorLight/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["WiFi_SSID",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["40008029","0405C80001000600003C5A0A002800000000","000080C0","00006000","00004000","00000000"]}}
15:30:36.533 MQT: stat/MonitorLight/STATUS4 = {"StatusMEM":{"ProgramSize":626,"Free":376,"Heap":22,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"14405E","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
15:30:36.538 MQT: stat/MonitorLight/STATUS5 = {"StatusNET":{"Hostname":"MonitorLight","IPAddress":"10.1.1.168","Gateway":"10.1.1.1","Subnetmask":"255.255.255.0","DNSServer1":"10.1.1.4","DNSServer2":"1.1.1.1","Mac":"60:01:94:66:B2:52","Webserver":2,"HTTP_API":1,"WifiConfig":2,"WifiPower":17.0}}
15:30:36.542 MQT: stat/MonitorLight/STATUS6 = {"StatusMQT":{"MqttHost":"mqtt.example.com","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_66B252","MqttUser":"homeassistant","MqttCount":45,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
15:30:36.548 MQT: stat/MonitorLight/STATUS7 = {"StatusTIM":{"UTC":"2022-12-02T14:30:36","Local":"2022-12-02T15:30:36","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"08:23","Sunset":"16:55"}}
15:30:36.553 MQT: stat/MonitorLight/STATUS10 = {"StatusSNS":{"Time":"2022-12-02T15:30:36"}}
15:30:36.559 MQT: stat/MonitorLight/STATUS11 = {"StatusSTS":{"Time":"2022-12-02T15:30:36","Uptime":"31T01:21:30","UptimeSec":2683290,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":45,"POWER":"OFF","Wifi":{"AP":1,"SSId":"WiFi_SSID","BSSId":"[WiFi_BSSID]","Channel":4,"Mode":"11n","RSSI":58,"Signal":-71,"LinkCount":39,"Downtime":"0T00:02:22"}}}
```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
14:35:38.093 MQT: stat/MonitorLight/RESULT = {"Time":"2022-12-05T14:35:38","Uptime":"34T00:26:32","UptimeSec":2939192,"Heap":7,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":52,"POWER":"OFF","Wifi":{"AP":1,"SSId":"[REDACTED WAP Name]","BSSId":"[REDACTED WAP BSSId]","Channel":4,"Mode":"11n","RSSI":48,"Signal":-76,"LinkCount":43,"Downtime":"0T00:02:40"}}
14:35:38.134 SRC: MQTT
14:35:38.136 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 2, Pld 11, Data '11'
14:35:38.144 MQT: stat/MonitorLight/STATUS11 = {"StatusSTS":{"Time":"2022-12-05T14:35:38","Uptime":"34T00:26:32","UptimeSec":2939192,"Heap":8,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":52,"POWER":"OFF","Wifi":{"AP":1,"SSId":"[REDACTED WAP Name]","BSSId":"[REDACTED WAP BSSId]","Channel":4,"Mode":"11n","RSSI":48,"Signal":-76,"LinkCount":43,"Downtime":"0T00:02:40"}}}
14:35:53.878 WIF: Checking connection...
14:36:14.116 WIF: Checking connection...
14:36:19.975 WIF: Sending Gratuitous ARP
14:36:34.364 WIF: Checking connection...
14:36:54.630 WIF: Checking connection...
14:37:14.857 WIF: Checking connection...
14:37:19.968 WIF: Sending Gratuitous ARP
14:37:35.115 WIF: Checking connection...
14:37:55.388 WIF: Checking connection...
14:38:15.649 WIF: Checking connection...
14:38:19.980 WIF: Sending Gratuitous ARP
14:38:36.151 WIF: Checking connection...
14:38:56.356 WIF: Checking connection...
14:39:16.878 WIF: Checking connection...
14:39:19.986 WIF: Sending Gratuitous ARP
14:39:36.897 WIF: Checking connection...
14:39:57.135 WIF: Checking connection...
14:40:08.974 MQT: tele/MonitorLight/STATE = {"Time":"2022-12-05T14:40:08","Uptime":"34T00:31:02","UptimeSec":2939462,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":52,"POWER":"OFF","Wifi":{"AP":1,"SSId":"[REDACTED WAP Name]","BSSId":"[REDACTED WAP BSSId]","Channel":4,"Mode":"11n","RSSI":48,"Signal":-76,"LinkCount":43,"Downtime":"0T00:02:40"}}
14:40:17.132 WIF: Checking connection...
14:40:19.985 WIF: Sending Gratuitous ARP
14:40:37.125 WIF: Checking connection...
14:40:57.140 WIF: Checking connection...
14:41:17.409 WIF: Checking connection...
14:41:19.976 WIF: Sending Gratuitous ARP
14:41:37.376 WIF: Checking connection...
14:41:57.367 WIF: Checking connection...
14:41:57.619 MQT: Attempting connection...
14:41:58.622 DNS: Unable to resolve 'mqtt.example.com'
14:41:58.624 MQT: Connect failed to mqtt.example.com:1883, rc -5. Retry in 10 sec
14:42:09.668 MQT: Attempting connection...
14:42:10.672 DNS: Unable to resolve 'mqtt.example.com'
14:42:10.674 MQT: Connect failed to mqtt.example.com:1883, rc -5. Retry in 20 sec
14:42:19.191 WIF: Checking connection...
14:42:19.993 WIF: Sending Gratuitous ARP
14:42:31.690 MQT: Attempting connection...
14:42:32.124 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 30 sec
14:42:39.448 WIF: Checking connection...
14:42:59.956 WIF: Checking connection...
14:43:03.182 MQT: Attempting connection...
14:43:03.397 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 40 sec
14:43:19.983 WIF: Sending Gratuitous ARP
14:43:20.183 WIF: Checking connection...
14:43:40.188 WIF: Checking connection...
14:43:44.445 MQT: Attempting connection...
14:43:44.659 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 50 sec
14:44:00.458 WIF: Checking connection...
14:44:19.959 WIF: Sending Gratuitous ARP
14:44:20.459 WIF: Checking connection...
14:44:35.687 MQT: Attempting connection...
14:44:35.899 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 60 sec
14:44:40.434 WIF: Checking connection...
14:45:00.454 WIF: Checking connection...
14:45:09.009 RSL: STATE = {"Time":"2022-12-05T14:45:09","Uptime":"34T00:36:03","UptimeSec":2939763,"Heap":13,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":38,"MqttCount":52,"POWER":"OFF","Wifi":{"AP":1,"SSId":"[REDACTED WAP Name]","BSSId":"[REDACTED WAP BSSId]","Channel":4,"Mode":"11n","RSSI":48,"Signal":-76,"LinkCount":43,"Downtime":"0T00:02:40"}}
14:45:19.974 WIF: Sending Gratuitous ARP
14:45:20.725 WIF: Checking connection...
14:45:36.936 MQT: Attempting connection...
14:45:37.387 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 70 sec
14:45:40.697 WIF: Checking connection...
14:46:00.950 WIF: Checking connection...
14:46:19.970 WIF: Sending Gratuitous ARP
14:46:20.963 WIF: Checking connection...
14:46:40.949 WIF: Checking connection...
14:46:48.189 MQT: Attempting connection...
14:46:48.270 MQT: Connected
14:46:48.275 MQT: tele/MonitorLight/LWT = Online (retained)
14:46:48.278 MQT: cmnd/MonitorLight/POWER = 
14:46:48.280 MQT: Subscribe to cmnd/MonitorLight/#
14:46:48.283 MQT: Subscribe to cmnd/tasmotas/#
14:46:48.285 MQT: Subscribe to cmnd/tasmota-bin/#
14:46:48.287 MQT: Subscribe to cmnd/DVES_66B252_fb/#
14:46:49.001 MQT: tasmota/discovery/60019466B252/config = {"ip":"10.1.1.168","dn":"Monitor Light","fn":["Monitor Light",null,null,null,null,null,null,null],"hn":"MonitorLight","mac":"60019466B252","md":"Sonoff Basic","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"12.2.0","t":"MonitorLight","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[1,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":1,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"sho":[0,0,0,0],"sht":[[0,0,48],[0,0,46],[0,0,110],[0,0,108]],"ver":1} (retained)
14:46:49.011 MQT: tasmota/discovery/60019466B252/sensors = {"sn":{"Time":"2022-12-05T14:46:49"},"ver":1} (retained)
14:46:49.391 SRC: MQTT
14:46:49.393 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 4, Pld -99, Data 'None'
14:46:49.400 MQT: stat/MonitorLight/STATUS = {"Status":{"Module":1,"DeviceName":"Monitor Light","FriendlyName":["Monitor Light"],"Topic":"MonitorLight","ButtonTopic":"0","Power":0,"PowerOnState":0,"LedState":0,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":1,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
```

### TO REPRODUCE
There are no steps to reproduce this other than to have a device with v12.2.0 and have a MQTT broker configured via its DNS name.

### EXPECTED BEHAVIOUR
Wi-fi drops happen, but TASMOTA should be able to resolve the name of, and reconnect to, the MQTT broker.

### ADDITIONAL CONTEXT
n/a

**Comment by barbudor** [comment]
> "MqttHost":"mqtt.example.com"

Are you using a broker on the Internet ?
Or is that a local one with some DNS trickery ?

Scrambling the requested information is not helping us to understand your setup


Everything starts by a DNS problem:
> 14:41:58.622 DNS: Unable to resolve 'mqtt.example.com'
14:41:58.624 MQT: Connect failed to mqtt.example.com:1883, rc -5. Retry in 10 sec
14:42:09.668 MQT: Attempting connection...
14:42:10.672 DNS: Unable to resolve 'mqtt.example.com'
14:42:10.674 MQT: Connect failed to mqtt.example.com:1883, rc -5. Retry in 20 sec

So I would assume something is wrong there. Your MQTT broker may be up but not the DNS

**Comment by SteveDinn** [comment]
Should have been explicitly clear before:
* I have pi-hole running on the same network as the tasmota devices.
* My DHCP server points all clients to the pi-hole instance as the DNS server.
* The name "mqtt.mydomain.com" resolves to a local IP via pi-hole (in fact, the MQTT broker and pi-hole are running on the same raspberry pi).

And like I said, I can hit the pi-hole instance and successfully look up the name at the same time as tasmota is having these issues. and not ALL my tasmota devices have this problem, despite them all being the same version and many of them being Sonoff Basics.

**Comment by MaceMoneta** [comment]
DNS2 is set to Cloudflare (1.1.1.1). If you remove that, does the problem still occur?

**Reply by SteveDinn** [reply]
I'll try this, but I'll have to wait a least a day to give you a definitive answer.

**Reply by SteveDinn** [reply]
Hrm...What value would "remove" the secondary DNS server? `127.0.0.1`? `0.0.0.0`?

**Reply by sfromis** [reply]
`0.0.0.0` should indeed work for that.

**Reply by SteveDinn** [reply]
Having tried setting my secondary DNS server to `0.0.0.0` for the past couple of days has yielded no change in this behaviour. I'm going to try setting my MQTT Host back to an IP address to ensure the behaviour no longer happens. If this turns out to be the case, I think there may be some DNS gremlins in TASMOTA, because all other devices on my network continue to use that Pi-Hole DNS server successfully.

TL;DR: It's always DNS.

**Reply by SteveDinn** [reply]
So, after a few days of having my MQTT Host as an IP address rather than a name, it's clear that DNS is an issue. The drops still happen from time to time, but the reconnection is pretty much instant.

What is still unclear is where the DNS problem lies. Everything uses DNS constantly and no other services are complaining. I'm skeptical that my Pi-Hole is at fault here. Any idea how we can determine what's going on?

**Comment by MaceMoneta** [comment]
Do you have a system with tcpdump? If so, you can monitor DNS activity to/from the device in question to see what's happening when the problem occurs.

**Reply by SteveDinn** [reply]
I've switched back to using the DNS name for the MQTT host and am running tcpdump now.
```
$ sudo tcpdump -i eth0 'host 10.1.1.4 and host 10.1.1.169 and (src port 53 or dst port 53)'
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
16:15:14.450571 IP 10.1.1.169.58948 > 10.1.1.4.53: 1+ A? pool.ntp.org. (30)
16:15:14.451889 IP 10.1.1.4.53 > 10.1.1.169.58948: 1 4/0/0 A 137.190.2.4, A 73.61.36.59, A 138.236.128.36, A 129.146.193.200 (94)
16:15:15.706647 IP 10.1.1.169.58948 > 10.1.1.4.53: 2+ A? mqtt.example.com. (30)
16:15:15.707827 IP 10.1.1.4.53 > 10.1.1.169.58948: 2* 1/0/0 A 10.1.1.4 (46)
16:15:15.712102 IP 10.1.1.169.58948 > 10.1.1.4.53: 3+ A? mqtt.example.com. (30)
16:15:15.712855 IP 10.1.1.4.53 > 10.1.1.169.58948: 3* 1/0/0 A 10.1.1.4 (46)
```

That's what results from TASMOTA's start-up sequence. Note that in my case, both DNS (Pi-Hole) and the MQTT broker (mosquitto) are running on 10.1.1.4.

I'll have to wait and see what transpires during a failure.

**Reply by SteveDinn** [reply]
Very interesting results. The entire time TASMOTA says that it's trying to contact the MQTT broker, there are NO calls to the DNS server until just before the time it succeeds.

One is local time and the other is UTC, but the minutes match up.

## From tcpdump
```
17:16:36.435857 IP 10.1.1.169.58948 > dns.example.com.53: 4+ A? pool.ntp.org. (30)
17:16:36.437329 IP dns.example.com.domain > 10.1.1.169.58948: 4 4/0/0 A 198.50.127.72, A 45.153.48.114, A 149.56.37.32, A 209.115.181.106 (94)
17:45:50.247886 IP 10.1.1.169.58948 > dns.example.com.53: 6+ A? mqtt.example.com. (30)
17:45:50.249042 IP dns.example.com.domain > 10.1.1.169.58948: 6* 1/0/0 A 10.1.1.4 (46)
17:45:50.254987 IP 10.1.1.169.58948 > dns.example.com.53: 7+ A? mqtt.example.com. (30)
17:45:50.255874 IP dns.example.com.domain > 10.1.1.169.58948: 7* 1/0/0 A 10.1.1.4 (46)
```

## From TASMOTA
```
22:43:11.396 WIF: Connecting to AP1 [WiFi-AP] Channel 4 BSSId [BSSId] in mode 11n as BedroomAmp...
22:43:11.401 MQT: Attempting connection...
22:43:12.001 MQT: Connect failed to mqtt.example.com:1883, rc -5. Retry in 10 sec
22:43:25.786 MQT: Attempting connection...
22:43:25.794 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 20 sec
22:43:46.790 MQT: Attempting connection...
22:43:46.803 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 30 sec
22:44:17.759 MQT: Attempting connection...
22:44:17.773 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 40 sec
22:44:58.780 MQT: Attempting connection...
22:44:58.794 MQT: Connect failed to mqtt.example.com:1883, rc -2. Retry in 50 sec
22:45:19.973 RSL: STATE = {"Time":"2022-12-12T22:45:19","Uptime":"0T01:30:11","UptimeSec":5411,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"[WiFi-AP]","BSSId":"[BSSId]","Channel":4,"Mode":"11n","RSSI":66,"Signal":-67,"LinkCount":2,"Downtime":"0T00:00:09"}}
22:45:49.791 MQT: Attempting connection...
22:45:49.823 MQT: Connected
22:45:49.826 MQT: tele/BedroomAmp/LWT = Online (retained)```

**Reply by barbudor** [reply]
Does Tasmota only have your local DNS in config ? No Internet fallback as you had some time back ?

I don't know what rule you have used in tcpdump but if you are looking exclusively at the exchange between tasmota and your local DNS, may be you don't see that Tasmota is trying to get the IP from the fallback DNS ?
Which could make sense if your local DNS had a failure earlier.

Note that the DNS resolution for the MQTT server is not done by Tasmota itself but from PubSubClient library calling socket API

Also you could try to check what tasmota is doing on port 1883 before the successfull connection. Is it trying to find the broker at another address ?

**Reply by barbudor** [reply]
@SteveDinn may be this PR https://github.com/arendst/Tasmota/pull/17387 could impact your issue as it changes who is performing the resolve for MQTT

NOT YET MERGED at this moment

**Reply by SteveDinn** [reply]
Right on. Thanks for that.

**Reply by s-hadinger** [reply]
This is merged now. I didn't see this issue but it looks indeed to be a similar problem. With Tasmota DNS resolver, there is no more caching of DNS. It should also be easier to troubleshoot.

**Comment by SteveDinn** [comment]
I ended up finally installing 12.3.1 with the DNS revolver change and my device was completely freaking out. WiFi scans couldn't find any access points, but despite this, it would sporadically connect for maybe a minute before dropping off for ~30 minutes at a time.,

I finally manually flashed to an empty bin before re-flashing it back to 12.2.0, and it was stable. So I OTA upgraded it back to 12.3.1 and it's been stable for the last hour. I now believe that it may have been something wonky with the device itself. It's been upgraded to most releases since v8.5, so who knows what was going on in there.
